WcaOnRails/app/javascript/edit-events/modals/Cutoff.jsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react'
+
+import events from 'wca/events.js.erb'
+import formats from 'wca/formats.js.erb'
+import AttemptResultInput from './AttemptResultInput'
+import {
+  pluralize,
+  matchResult,
+  roundIdToString,
+  attemptResultToString,
+} from './utils'
+
+export default {
+  Title({ wcifRound }) {
+    return <span>Cutoff for {roundIdToString(wcifRound.id)}</span>;
+  },
+  Show({ value: cutoff, wcifEvent }) {
+    let str;
+    if(cutoff) {
+      str = `Best of ${cutoff.numberOfAttempts} ${matchResult(cutoff.attemptResult, wcifEvent.id, { short: true })}`;
+    } else {
+      str = "-";
+    }
+    return <span>{str}</span>;
+  },
+  Input({ value: cutoff, onChange, autoFocus, wcifEvent, roundNumber }) {
+    let wcifRound = wcifEvent.rounds[roundNumber - 1];
+
+    let numberOfAttemptsInput, attemptResultInput;
+    let onChangeAggregator = () => {
+      let numberOfAttempts = parseInt(numberOfAttemptsInput.value);
+      let newCutoff;
+      if(numberOfAttempts > 0) {
+        newCutoff = {
+          numberOfAttempts,
+          attemptResult: attemptResultInput ? parseInt(attemptResultInput.value) : 0,
+        };
+      } else {
+        newCutoff = null;
+      }
+      onChange(newCutoff);
+    };
```

Comment:
Ah, so I'm not a big fan of multiline ternaries. I do like your idea to check for `=== 0` rather than `> 0` here, though, so I'm goiug to do that and then flip the `if...else` blocks.